benchmarks/src/bin/tpch.rs

```diff
@@ -549,6 +555,55 @@ mod tests {
         Ok(())
     }
 
+    struct Line(Option<usize>);
+
+    impl fmt::Display for Line {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            match self.0 {
+                None => write!(f, "    "),
+                Some(idx) => write!(f, "{:<4}", idx + 1),
+            }
+        }
+    }
+
+    fn assert_text_eq(expected: String, actual: String) {
```

Comment:
In general I think it would help a lot to remove tests from the benchmarks -- @liurenjie1024 is making good progress in this area with  https://github.com/apache/arrow-datafusion/pull/6435

Maybe we could avoid adding this extra comparison to the `tpch` binary and instead work on getting the tests out of the benchmark runner in the first place 🤔